src/fhirtypes/ElementDefinition.ts

```diff
@@ -1642,8 +1643,16 @@ export class ElementDefinition {
    * @throws {InvalidUriError} when the system being assigned is not a valid uri
    */
   private assignFshCode(code: FshCode, exactly = false, fisher?: Fishable): void {
-    if (code.system && !isUri(code.system.split('|')[0])) {
-      throw new InvalidUriError(code.system);
+    if (code.system) {
+      const csURI = code.system.split('|')[0];
+      if (fisher) {
+        const vsURI = fisher.fishForMetadata(code.system, Type.ValueSet)?.url.split('|')[0] ?? '';
+        if (isUri(vsURI)) {
```

Comment:
Is it necessary to check if the `vsURI` is a URI here? I can't figure out why we need to do that. We are getting a `url`, so it is either going to exist or not, but if it exists won't it always be a `url`? Isn't the question here more if the input `code.system` resolves to a ValueSet? I don't think we need to then check that the ValueSet has a valid URI, but I could be missing something here.

Reply:
I guess it won't always necessarily be a value URI, because a user could use a caret rule to set it to something invalid. But I'm still a bit unsure about if we need to actually be checking if these are URIs or not.

Reply:
No you're totally right! I could just check if the `vsURI` is truthy or not, rather than checking if the URI is valid.

Reply:
Addressed in `e8b6e3f`.